xonsh/pyghooks.py

```diff
@@ -1596,7 +1596,10 @@ def color_file(file_path: str, path_stat: os.stat_result) -> tuple[_TokenType, s
 
 
 def _command_is_valid(cmd):
-    return (cmd in XSH.aliases or locate_executable(cmd)) and not iskeyword(cmd)
+    if XSH.env.get("XONSH_PROMPT_HIGHLIGHT_EXECUTABLE"):
+        return cmd in XSH.aliases and not iskeyword(cmd)
+    else:
+        return (cmd in XSH.aliases or locate_executable(cmd)) and not iskeyword(cmd)
```

Comment:
The performance and recommendation of  `executables.locate_executable()` vs `environ.locate_binary()` needs to be reviewed, as while `environ.locate_binary()` is not recommended `environ.locate_binary()` per:
> NOT RECOMMENDED because ``commands_cache.locate_binary`` contains ``update_cache``
    with scanning all files in ``$PATH``.
    
it does not look like this is actually the case anymore.